### PR TITLE
[BEAM-11589] Migrated release build.gradle file to Kotlin script

### PR DESCRIPTION
**Pull request by ramazan-yapparov**
Migrated release build.gradle file to Kotlin script

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by ramazan-yapparov** [comment]
R: @TheNeuralBit

**Comment by ramazan-yapparov** [comment]
@TheNeuralBit 
I can move the `libraries` map from Groovy module to Kotlin file in `buildSrc` project and change is from being a `Map` to `object`
That way we will be able to access that object without this ugly casts and use object fields access instead of some "magic string"

https://github.com/apache/beam/blob/022c7943df9817347bca6a6ed74cef90711ba47e/release/build.gradle.kts#L28-L33

**Comment by ramazan-yapparov** [comment]
Run Python PreCommit

**Comment by ramazan-yapparov** [comment]
Run Java PreCommit

**Review by TheNeuralBit — APPROVED** [review]
LGTM, thank you!

> I can move the libraries map from Groovy module to Kotlin file in buildSrc project and change is from being a Map to object
That way we will be able to access that object without this ugly casts and use object fields access instead of some "magic string"

Are you saying you can do this once BeamModulePlugin.groovy has been converted? If so please add a TODO to remind us to follow up with that

**Comment by ramazan-yapparov** [comment]
> LGTM, thank you!
> 
> > I can move the libraries map from Groovy module to Kotlin file in buildSrc project and change is from being a Map to object
> > That way we will be able to access that object without this ugly casts and use object fields access instead of some "magic string"
> 
> Are you saying you can do this once BeamModulePlugin.groovy has been converted? If so please add a TODO to remind us to follow up with that

No, I can do it now - move only `libraries` and leave the rest as is

**Comment by TheNeuralBit** [comment]
> No, I can do it now - move only libraries and leave the rest as is

Oh I see. Sure let's go ahead and do that, assuming you can do it without changing all the existing `libraries` usages.

File: buildSrc/src/main/kotlin/org/apache/beam/gradle/BeamDependencies.kt [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    object groovy {
+        const val groovy_all = "org.codehaus.groovy:groovy-all:2.4.13"
+    }
+    object maven {
+        const val maven_compiler_plugin = "maven-plugins:maven-compiler-plugin:3.7.0"
+        const val maven_exec_plugin     = "maven-plugins:maven-exec-plugin:1.6.0"
+        const val maven_jar_plugin      = "maven-plugins:maven-jar-plugin:3.0.2"
+        const val maven_shade_plugin    = "maven-plugins:maven-shade-plugin:3.1.0"
+        const val maven_surefire_plugin = "maven-plugins:maven-surefire-plugin:3.0.0-M5"
+    }
```
Comment (TheNeuralBit) [inline buildSrc/src/main/kotlin/org/apache/beam/gradle/BeamDependencies.kt:252]: Is it possible to also remove `libraries` from `BeamModulePlugin.groovy`?

If the only way to do this is to duplicate the definitions, I don't think we should. These dependencies change quite often so the two definitions would start diverging.

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -21,6 +21,7 @@ plugins {
   `java-gradle-plugin`
   groovy
   id("com.diffplug.spotless") version "5.6.1"
+  `kotlin-dsl`
```
Comment (TheNeuralBit) [inline buildSrc/build.gradle.kts:24]: Is this a mistake?

**Review by TheNeuralBit — COMMENTED** [review]
Is it possible to move `libraries`? It looks like right now you've copied it which is risky.

File: buildSrc/src/main/kotlin/org/apache/beam/gradle/BeamDependencies.kt (hunk 1 above)
Comment (ramazan-yapparov) [reply buildSrc/src/main/kotlin/org/apache/beam/gradle/BeamDependencies.kt]: Right now the only way is to duplicate the definitions.
I guess we should move dependencies declaration to Kotlin at the same time as moving `BeamModulePlugin`

File: buildSrc/build.gradle.kts (hunk 2 above)
Comment (ramazan-yapparov) [reply buildSrc/build.gradle.kts]: It wasn't
I added this dependency because I moved dependencies to Kotlin in `buildSrc` module

**Comment by ramazan-yapparov** [comment]
Run Java PreCommit

File: buildSrc/src/main/kotlin/org/apache/beam/gradle/BeamDependencies.kt (hunk 1 above)
Comment (TheNeuralBit) [reply buildSrc/src/main/kotlin/org/apache/beam/gradle/BeamDependencies.kt]: ok thanks, sorry about that.

**Review by TheNeuralBit — APPROVED** [review]

**Pull request merged**